components/collator/src/options.rs

```diff
@@ -296,10 +337,44 @@ pub enum MaxVariable {
 ///
 /// Variable top (BCP47 `vt`) is unsupported (use Max Variable instead). `vt`
 /// is deprecated and prohibited by ECMA 402.
-#[derive(Copy, Clone, Debug)]
-pub struct CollatorOptions(u32);
+#[non_exhaustive]
+#[derive(Debug, Copy, Clone)]
+pub struct CollatorOptions {
+    /// User-specified strength collation option.
+    pub strength: Option<Strength>,
+    /// User-specified alternate handling collation option.
+    pub alternate_handling: Option<AlternateHandling>,
+    /// User-specified case first collation option.
+    pub case_first: Option<CaseFirst>,
+    /// User-specified max variable collation option.
+    pub max_variable: Option<MaxVariable>,
+    /// User-specified case level collation option.
+    pub case_level: Option<CaseLevel>,
+    /// User-specified numeric collation option.
+    pub numeric: Option<Numeric>,
+    /// User-specified backward second level collation option.
+    pub backward_second_level: Option<BackwardSecondLevel>,
+}
 
 impl CollatorOptions {
+    /// Create a new `CollatorOptions` with the defaults.
+    pub const fn new() -> Self {
+        Self {
+            strength: Some(Strength::Tertiary),
```

Comment:
Even this should probably be `None` in case CLDR ever wants to default to quaternary for Japanese. (Right now, it seems that CLDR doesn't want to do that, but it would be consistent with everything else to handle the strength the same way as the rest. Right now, we also don't have a mechanism for reading the strength default from locale data, because no locale wants it. The Japanese locale states the tertiary level explicitly, though.)

Reply:
Done.